sympy/solvers/ode/tests/test_ode.py

```diff
@@ -2883,6 +2883,43 @@ def test_separable_reduced():
     assert sol == Eq(log(x**2*f(x))/2 - log(x**2*f(x) - 2)/2, C1 + log(x))
     assert checkodesol(eq, sol, order=1, solve_for_func=False)[0]
 
+    eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x**(S(2)/3)*f(x))**2), 0)
+    # FIXME: This one hangs
+    #sol = dsolve(eq, hint = 'seprable_reduced')
+    # assert checkodesol(eq, sol) = (True, 0)
```

Comment:
This should link to the associated issue.

This also works with `simplify=False` so that should be tested as well:
```julia
In [67]: eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x**(S(2)/3)*f(x))**2), 0)                                                          

In [68]: sol = dsolve(eq, simplify=False)                                                                                         

In [69]: sol                                                                                                                      
Out[69]: 
⎡              __________________               __________________⎤
⎢             ╱        2/3                     ╱        2/3       ⎥
⎢            ╱        x                       ╱        x          ⎥
⎢f(x) = -   ╱   ──────────────── , f(x) =    ╱   ──────────────── ⎥
⎢          ╱          8/3      2            ╱          8/3      2 ⎥
⎣        ╲╱     2⋅C₁⋅x    - 3⋅x           ╲╱     2⋅C₁⋅x    - 3⋅x  ⎦

In [70]: checkodesol(eq, sol, solve_for_func=False)                                                                               
Out[70]: [(True, 0), (True, 0)]
```